### ISSUE TITLE
Change in parallel frame handling for 0.9.3 causes 'overfull content'

### ISSUE THREAD

**Issue by alerque** [simoncozens/sile#181]
Something in between the 0.9.2 and 0.9.3 changed in the way framesets are handled and I can't get my head around where my error is coming from:

> ! Overfull content for frame supurge

I'm attaching sources ([gorevler.sil](https://github.com/simoncozens/sile/files/11605/gorevler.sil.txt),  [chart.lua](https://github.com/simoncozens/sile/files/11606/chart.lua.txt)) for a document I'm testing. I'm sure there is a better way to do this now. This is ugly because it started out as a test of tables (which I couldn't get to work) and ended up as frames just to see how it would go and it turned into a way to typeset a CSV data file into a signup chart. The expected output of this data is a single page fairly compact list. Currently for some reason extra line breaks are getting added in each row pushing everything into 2 pages (+ an blank at the end of the document?) and the above error. The extra whitespace (highlighted red) wasn't being added in the last version of sile:

![gorevler pdf_072](https://cloud.githubusercontent.com/assets/173595/10424655/96334470-70d9-11e5-8647-53ed441f79e1.png)

I've tried removing various elements from this, blanking the data so there can't be wrapping issues, removing columns, etc. It seems like whatever the last frame with output is before `\sync` is called throws the error. In the default case this is 'supurge', but if you remove any output to that frame in becomes 's', the one before it and so on.

What's going on here with the parallel frames system that is different from the previous release (and what would be a saner way to typeset something like this)?

**Comment by alerque** [comment]
Okay so the overfull error seems to be unrelated to the whitespace issue. Removing these two lines (relic of an early attempt to lay this out?) fixes the latter problem:

    \set[parameter=document.parskip,value=32pt]
    \set[parameter=document.baselineskip,value=30pt]

The overfull content error still displays but nothing actually seems broken except that a trailing blank page is added to the document ([current PDF output](https://owncloud.alerque.com/index.php/s/JipKlZ7bR43zToL)). Clearly this isn't the right way to go about typesetting this, but I'm still not sure what is.

**Comment by simoncozens** [comment]
I think what is happening is that parskip is only being added to the typesetter (the column) which is active at the end of the paragraph. Basically paragraphs within a parallel mode don't make much sense anyway. Try the latest version, which ignores items added between the call to `\sync` and setting up the first content frame.

**Comment by neofob** [comment]
As of ca55d1c2 commit, this is still a problem.

**Comment by simoncozens** [comment]
Hmmm. This is the output I get from the attached files:

![screen shot 2015-11-08 at 14 45 12](https://cloud.githubusercontent.com/assets/106728/11019013/5775a990-8627-11e5-9e5f-a1673e49b0fe.png)

I also get copious debugging output, which I've now removed. What do you get? (And can you show me the debugging output?)

**Comment by alerque** [comment]
At the current HEAD commit of 0.9.4-unreleased, I get:

> Error detected:  
> core/harfbuzz-shaper.lua:27: bad argument #6 to '_shape' (number expected, got string)

So that's not too useful. With 0.9.3 release version I get this output ([debug.txt](https://github.com/simoncozens/sile/files/29484/debug.txt)) from the above attached input files and without removing the parskip value as per my comment:

![selection_076](https://cloud.githubusercontent.com/assets/173595/11019413/b6b60804-8600-11e5-8ab1-25320df4b760.png)

**Comment by simoncozens** [comment]
I have *half* an idea where it's coming from. Notably if there's no linebreak after the circled number, it's fine:

    \b {①} \bulasik {Anya \skip[height=10pt]}
    \t {①} \temizlik {Janna \skip[height=10pt]}
    \s {①} \supurge {Nathan \skip[height=10pt]}

But why that is, I don't know.

**Comment by neofob** [comment]
@simoncozens : I now have this error with newly compiled `sile` from master (as of 613f01f commit).
```
This is SILE 0.9.4-unreleased
/usr/bin/lua: /usr/local/share/sile/core/sile.lua:184: attempt to index field 'masterFilename' (a nil value)
stack traceback:
	/usr/local/share/sile/core/sile.lua:184: in function 'resolveFile'
	/usr/local/share/sile/core/sile.lua:64: in function 'require'
	/usr/local/share/sile/core/harfbuzz-shaper.lua:5: in main chunk
	[C]: in function 'require'
	/usr/local/share/sile/core/sile.lua:50: in function 'init'
	/usr/local/bin/sile:11: in main chunk
	[C]: in ?
```

**Comment by alerque** [comment]
@neofob I am unable to replicate that problem. I just compiled the same commit and was able to build both the sample source from this issue and the documentation without a problem. Perhaps you could give a little bore detail about the steps to reproduce in a new issue? Specifically what commands you are running on what input files would be helpful to know.

**Comment by simoncozens** [comment]
@alerque Last time I saw your error it was related to regenerating sile from sile.in. Rerun configure?

**Comment by simoncozens** [comment]
Uh, that should be @neofob

**Comment by deepakjois** [comment]
I am running into this error as well, even after rerunning configure.

Actually the problem is this line in the sile executable script:

```
  SILE.masterFilename = unparsed[1]:gsub("\\", "/")
```
Which comes _after_ `SILE.init` is already called.

This should be located inside the `SILE.parseArguments` function instead, so that when `SILE.init` is called in the sile executable script, it is already assigned.

**Comment by alerque** [comment]
@deepakjois @neofob Are you guys executing your custom compiled SILE binary from outside the repo root directory (i.e. with a hard coded path)?

**Comment by deepakjois** [comment]
Yes, I am executing the SILE binary from outside the repo. But not with a hardcoded path (the SILE binary is inside one of my PATH folders). I ran `make install`. But I am also hacking on the code right now, so I run `make install` every time I make a change, and then run the SILE binary from PATH.

**Comment by neofob** [comment]
@simoncozens , @alerque : It could be it. I recompiled the source code with new commit.
```
$ git pull
$ make -j4
as root
# make install
```

The sile files are at @ https://github.com/neofob/ebooks.neofob/tree/Vindication/AVOROW/en
The above error was the result of running sile command at `../en` directory
```
$ sile -I requirements.sil ch-04_en.sil
```
